### PR TITLE
SDIT-1491: ✨ Add in enum to constrain attributes

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResource.kt [hunk 1]
```diff
@@ -26,20 +27,19 @@ import uk.gov.justice.digital.hmpps.prisonersearch.search.services.ReferenceData
 class ReferenceDataResource(private val referenceDataService: ReferenceDataService) {
   @Operation(
     summary = "*** BETA *** Reference data search",
-    description = """
-      BETA endpoint - reference data returned reflects the data assigned to prisoners rather than all the possible
+    description = """BETA endpoint - reference data returned reflects the data assigned to prisoners rather than all the possible
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResource.kt:30]: swagger issue - if start with a newline then it things the description is code and puts it in purple instead 🤷

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResource.kt [hunk 2]
```diff
@@ -26,20 +27,19 @@ import uk.gov.justice.digital.hmpps.prisonersearch.search.services.ReferenceData
...
       values.  Only to be used for searching existing data purposes.
       """,
     security = [SecurityRequirement(name = "global-search-role"), SecurityRequirement(name = "prisoner-search-role")],
     responses = [
       ApiResponse(
         responseCode = "200",
-        description = "Search successfully performed",
+        description = "Reference data search successfully performed",
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResource.kt:37]: improvements from last PR

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt [hunk 3]
```diff
@@ -19,59 +18,59 @@ class ReferenceDataService(
...
 
     return try {
       val searchResponse = elasticsearchClient.search(searchRequest)
-      val aggregation: MultiBucketsAggregation = searchResponse.aggregations.get(convertedAttribute)
+      val aggregation: MultiBucketsAggregation = searchResponse.aggregations.get(attribute.name)
       ReferenceDataResponse(
         aggregation.buckets.map {
           ReferenceData(it.keyAsString)
-        },
+        }.sortedBy { it.key },
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt:31]: add sorting of results too

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt [hunk 4]
```diff
@@ -19,59 +18,59 @@ class ReferenceDataService(
...
     timeout(TimeValue(searchTimeoutSeconds, TimeUnit.SECONDS))
     size(0)
-    aggregation(TermsAggregationBuilder(attribute).size(maxSearchResults).field(attribute))
+    aggregation(TermsAggregationBuilder(attribute.name).size(maxSearchResults).field(attribute.field))
   }
 
   private companion object {
     private val log: Logger = LoggerFactory.getLogger(this::class.java)
-    private val attributeMappings = mapOf(
-      "build" to "build.keyword",
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt:53]: don't need the full list here hardcoded now

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResourceTest.kt [hunk 5]
```diff
@@ -190,21 +193,24 @@ class ReferenceDataResourceTest : AbstractSearchDataIntegrationTest() {
 
   private fun attributeData(): Stream<Arguments> =
     Stream.of(
-      arguments("build", listOf("Proportional", "Muscular", "Obese")),
-      arguments("ethnicity", listOf("White: Any other background", "Prefer not to say")),
+      arguments("build", listOf("Muscular", "Obese", "Proportional")),
+      // arguments("category", listOf("White: Any other background", "Prefer not to say")),
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResourceTest.kt:197]: will do these two next

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResourceTest.kt [hunk 6]
```diff
@@ -190,21 +193,24 @@ class ReferenceDataResourceTest : AbstractSearchDataIntegrationTest() {
...
+      // arguments("csra", listOf("White: Any other background", "Prefer not to say")),
+      arguments("ethnicity", listOf("Prefer not to say", "White: Any other background")),
       arguments("facialHair", listOf("Clean Shaven", "Goatee Beard", "Not Asked")),
       arguments("gender", listOf("Female", "Male", "Not Known / Not Recorded")),
-      arguments("hairColour", listOf("Red", "Balding", "Mouse")),
+      arguments("hairColour", listOf("Balding", "Mouse", "Red")),
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/ReferenceDataResourceTest.kt:202]: had to change these all now to be sorted

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**